PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/helpers/CustomerSurveyHelper.java

```diff
@@ -65,14 +66,19 @@ public void showFeedbackNotification(Project project) {
             Observable.timer(POP_UP_DELAY, TimeUnit.SECONDS).subscribeOn(Schedulers.io())
                     .take(1)
                     .subscribe(next -> {
-                        SurveyPopUpDialog dialog = new SurveyPopUpDialog(CustomerSurveyHelper.this, project);
-                        dialog.setVisible(true);
-                        synchronized (CustomerSurveyHelper.class) {
-                            if (operation != null) {
-                                operation.complete();
+                        try {
+                            SurveyPopUpDialog dialog = new SurveyPopUpDialog(CustomerSurveyHelper.this, project);
+                            dialog.setVisible(true);
+                            synchronized (CustomerSurveyHelper.class) {
+                                if (operation != null) {
+                                    operation.complete();
+                                }
+                                operation = TelemetryManager.createOperation(SYSTEM, SURVEY);
+                                operation.start();
                             }
-                            operation = TelemetryManager.createOperation(SYSTEM, SURVEY);
-                            operation.start();
+                        } catch (Exception e) {
+                            // Survey exceptions should not block user
+                            EventUtil.logError(SYSTEM, SURVEY, ErrorType.systemError, e, null, null);
                         }
                     });
```

Comment:
It might be better to add a error handler in subscribe.

Reply:
Nice idea, fixed.